### PR TITLE
CM-552: How to Run an Async Method Synchronously in .NET

### PR DESCRIPTION
**Pull request by aineaTheSecond**

### CODE REVIEW

**Review by karimkod — CHANGES_REQUESTED** [review]

File: async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public List<Person> GetPeople()
+    {
+        var task = new Task<List<Person>>(() => new List<Person>
+        {
+            new() { Name = "Alice", Age = 25 },
+            new() { Name = "Bob", Age = 30 },
+            new() { Name = "Charlie", Age = 35 }
+        });
+
+        Thread.Sleep(5000);
```
[the remark this answers is not in the document]
Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:14]: I've updated my explanation to better reflect what is in the source code. Please let me know if that will suffice.

[the remark this answers is not in the document]
Comment (karimkod) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:14]: It still doesn't reflect what's explained in the article. I've updated the review about this. you can check it.

[the remark this answers is not in the document]
Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs]: This is done. Please take a look.

**Review by karimkod — CHANGES_REQUESTED** [review]

File: async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            new() { Name = "Bob", Age = 30 },
+            new() { Name = "Charlie", Age = 35 }
+        };
+    }
+
+    public async Task ThrowExceptionAsync()
+    {
+        await Task.Delay(1000);
+
+        throw new Exception("The task threw an exception");
```
[the remark this answers is not in the document]
Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs]: I've made the update, please take a look

File: async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    public void ExceptionHandlingUsingGetAwaiterMethod()
+    {
+        var task = ThrowExceptionAsync();
+        try
+        {
+            task.GetAwaiter().GetResult();
+        }
+        catch (Exception e)
```
Comment (karimkod) [inline async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:92]: This too should change to InvalidOperationException.

**Review by karimkod — CHANGES_REQUESTED** [review]

File: async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs (hunk 3 above)
Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs]: Done

File: async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace RunAsyncMethodSynchronously;
+
+public class PersonService
+{
+    public List<Person> GetPeople()
+    {
+        var task = new Task<List<Person>>(() =>
+        {
+            Thread.Sleep(1000);
```
Comment (karimkod) [inline async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:11]: For all delays in the Github code, I know you have reduced them to be 1 second but they can be reduced further to 200 (2ms) for instance. 
### **Please, do this just for the Code in Github.  The code in the article should stay with 1 second.**

**Review by karimkod — CHANGES_REQUESTED** [review]

File: async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs (hunk 4 above)
Comment (karimkod) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:11]: Sorry @aineaTheSecond, One more change and we will be good :)

Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:11]: No problem, changes coming up in a minute

Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs:11]: So the code in the article should be different from what's in the source code on GitHub?

Comment (karimkod) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs]: Thanks! 
Yes, for this, it's fine. 
We don't want the Nightly build to take a lot of time and I think the readers will get the point.

Comment (aineaTheSecond) [reply async-csharp/RunAsycMethodSynchronously/RunAsycMethodSynchronously/PersonService.cs]: Great!
If there are more changes to be made, just let me know and I'll make the updates.
Thanks!

**Comment by aineaTheSecond** [comment]
> Hello @aineaTheSecond, I've noticed that the folder name has a typo can you fix that as well as the solution file name.

I had not noticed before, but I've made the change. Please check. 
Thanks.

**Pull request merged**